components/style/binding_tools/check_bindings.py

```diff
@@ -0,0 +1,27 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import re
+
+
+def build(objdir, verbose=False):
+    with open(objdir + "/bindings.rs", "r") as bindings:
+        with open(objdir + "/check_bindings.rs", "w+") as tests:
```

Comment:
Instead of nesting inside, this could be joined as

``` python
with open(...) as bindings, open(...) as tests:
```

Also, as @aneeshusa said, it's a better idea to join the paths instead of hardcoding them, and put the paths in the top of the file, like `BINDINGS_FILE`, `BINDINGS_CHECK_FILE`, etc.